File: Ocis.Server/Ocis.Protocol/ProtocolSpec.ts
import { Record } from "./ocis-protocol-ts/fable_modules/fable-library-ts.4.25.0/Types.js";
import { int32, uint8, uint32 } from "./ocis-protocol-ts/fable_modules/fable-library-ts.4.25.0/Int32.js";
import { Option } from "./ocis-protocol-ts/fable_modules/fable-library-ts.4.25.0/Option.js";
import { IComparable, IEquatable } from "./ocis-protocol-ts/fable_modules/fable-library-ts.4.25.0/Util.js";
import { string_type, record_type, option_type, array_type, enum_type, int32_type, uint8_type, uint32_type, TypeInfo } from "./ocis-protocol-ts/fable_modules/fable-library-ts.4.25.0/Reflection.js";

export class RequestPacket extends Record implements IEquatable<RequestPacket>, IComparable<RequestPacket> {
    readonly MagicNumber: uint32;
    readonly Version: uint8;
    readonly CommandType: int32;
    readonly TotalPacketLength: int32;
    readonly KeyLength: int32;
    readonly ValueLength: int32;
    readonly Key: uint8[];
    readonly Value: Option<uint8[]>;
    constructor(MagicNumber: uint32, Version: uint8, CommandType: int32, TotalPacketLength: int32, KeyLength: int32, ValueLength: int32, Key: uint8[], Value: Option<uint8[]>) {
        super();
        this.MagicNumber = MagicNumber;
        this.Version = Version;
        this.CommandType = (CommandType | 0);
        this.TotalPacketLength = (TotalPacketLength | 0);
        this.KeyLength = (KeyLength | 0);
        this.ValueLength = (ValueLength | 0);
        this.Key = Key;
        this.Value = Value;
    }
}

export function RequestPacket_$reflection(): TypeInfo {
    return record_type("Ocis.Server.ProtocolSpec.RequestPacket", [], RequestPacket, () => [["MagicNumber", uint32_type], ["Version", uint8_type], ["CommandType", enum_type("Ocis.Server.ProtocolSpec.CommandType", int32_type, [["Set", 1], ["Get", 2], ["Delete", 3]])], ["TotalPacketLength", int32_type], ["KeyLength", int32_type], ["ValueLength", int32_type], ["Key", array_type(uint8_type)], ["Value", option_type(array_type(uint8_type))]]);
}

export class ResponsePacket extends Record implements IEquatable<ResponsePacket>, IComparable<ResponsePacket> {
    readonly MagicNumber: uint32;
    readonly Version: uint8;
    readonly StatusCode: uint8;
    readonly TotalPacketLength: int32;
    readonly ValueLength: int32;
    readonly ErrorMessageLength: int32;
    readonly Value: Option<uint8[]>;
    readonly ErrorMessage: Option<string>;
    constructor(MagicNumber: uint32, Version: uint8, StatusCode: uint8, TotalPacketLength: int32, ValueLength: int32, ErrorMessageLength: int32, Value: Option<uint8[]>, ErrorMessage: Option<string>) {
        super();
        this.MagicNumber = MagicNumber;
        this.Version = Version;
        this.StatusCode = StatusCode;
        this.TotalPacketLength = (TotalPacketLength | 0);
        this.ValueLength = (ValueLength | 0);
        this.ErrorMessageLength = (ErrorMessageLength | 0);
        this.Value = Value;
        this.ErrorMessage = ErrorMessage;
    }
}

export function ResponsePacket_$reflection(): TypeInfo {
    return record_type("Ocis.Server.ProtocolSpec.ResponsePacket", [], ResponsePacket, () => [["MagicNumber", uint32_type], ["Version", uint8_type], ["StatusCode", enum_type("Ocis.Server.ProtocolSpec.StatusCode", uint8_type, [["Success", 0], ["NotFound", 1], ["Error", 2]])], ["TotalPacketLength", int32_type], ["ValueLength", int32_type], ["ErrorMessageLength", int32_type], ["Value", option_type(array_type(uint8_type))], ["ErrorMessage", option_type(string_type)]]);
}

File: Ocis.Server/Ocis.Protocol/ProtocolSpec.dart
// ignore_for_file: camel_case_types, constant_identifier_names, non_constant_identifier_names, unnecessary_this
import './ocis-protocol-dart/fable_modules/fable_library/Types.dart' as types;
import './ocis-protocol-dart/fable_modules/fable_library/Util.dart' as util;

class RequestPacket implements types.Record, Comparable<RequestPacket> {
    final int MagicNumber;
    final int Version;
    final int CommandType;
    final int TotalPacketLength;
    final int KeyLength;
    final int ValueLength;
    final List<int> Key;
    final types.Some<List<int>>? Value;
    const RequestPacket(this.MagicNumber, this.Version, this.CommandType, this.TotalPacketLength, this.KeyLength, this.ValueLength, this.Key, this.Value);
    @override
    bool operator ==(Object other) => (other is RequestPacket) && ((other.MagicNumber == MagicNumber) && ((other.Version == Version) && ((other.CommandType == CommandType) && ((other.TotalPacketLength == TotalPacketLength) && ((other.KeyLength == KeyLength) && ((other.ValueLength == ValueLength) && (util.equalsList(other.Key, Key, (int x, int y) => x == y) && (other.Value == Value))))))));
    @override
    int get hashCode => util.combineHashCodes([MagicNumber.hashCode, Version.hashCode, CommandType.hashCode, TotalPacketLength.hashCode, KeyLength.hashCode, ValueLength.hashCode, Key.hashCode, Value.hashCode]);
    @override
    int compareTo(RequestPacket other) {
        late int $r;
        if (($r = MagicNumber.compareTo(other.MagicNumber)) == 0) {
            if (($r = Version.compareTo(other.Version)) == 0) {
                if (($r = CommandType.compareTo(other.CommandType)) == 0) {
                    if (($r = TotalPacketLength.compareTo(other.TotalPacketLength)) == 0) {
                        if (($r = KeyLength.compareTo(other.KeyLength)) == 0) {
                            if (($r = ValueLength.compareTo(other.ValueLength)) == 0) {
                                if (($r = util.compareList(Key, other.Key, (int x, int y) => x.compareTo(y))) == 0) {
                                    $r = util.compareNullable(Value, other.Value, (types.Some<List<int>> x, types.Some<List<int>> y) => x.compareTo(y));
                                }
                            }
                        }
                    }
                }
            }
        }
        return $r;
    }
}

class ResponsePacket implements types.Record, Comparable<ResponsePacket> {
    final int MagicNumber;
    final int Version;
    final int StatusCode;
    final int TotalPacketLength;
    final int ValueLength;
    final int ErrorMessageLength;
    final types.Some<List<int>>? Value;
    final types.Some<String>? ErrorMessage;
    const ResponsePacket(this.MagicNumber, this.Version, this.StatusCode, this.TotalPacketLength, this.ValueLength, this.ErrorMessageLength, this.Value, this.ErrorMessage);
    @override
    bool operator ==(Object other) => (other is ResponsePacket) && ((other.MagicNumber == MagicNumber) && ((other.Version == Version) && ((other.StatusCode == StatusCode) && ((other.TotalPacketLength == TotalPacketLength) && ((other.ValueLength == ValueLength) && ((other.ErrorMessageLength == ErrorMessageLength) && ((other.Value == Value) && (other.ErrorMessage == ErrorMessage))))))));
    @override
    int get hashCode => util.combineHashCodes([MagicNumber.hashCode, Version.hashCode, StatusCode.hashCode, TotalPacketLength.hashCode, ValueLength.hashCode, ErrorMessageLength.hashCode, Value.hashCode, ErrorMessage.hashCode]);
    @override
    int compareTo(ResponsePacket other) {
        late int $r;
        if (($r = MagicNumber.compareTo(other.MagicNumber)) == 0) {
            if (($r = Version.compareTo(other.Version)) == 0) {
                if (($r = StatusCode.compareTo(other.StatusCode)) == 0) {
                    if (($r = TotalPacketLength.compareTo(other.TotalPacketLength)) == 0) {
                        if (($r = ValueLength.compareTo(other.ValueLength)) == 0) {
                            if (($r = ErrorMessageLength.compareTo(other.ErrorMessageLength)) == 0) {
                                if (($r = util.compareNullable(Value, other.Value, (types.Some<List<int>> x, types.Some<List<int>> y) => x.compareTo(y))) == 0) {
                                    $r = util.compareNullable(ErrorMessage, other.ErrorMessage, (types.Some<String> x, types.Some<String> y) => x.compareTo(y));
                                }
                            }
                        }
                    }
                }
            }
        }
        return $r;
    }
}

